src/GitHub.VisualStudio/Services/ConnectionManager.cs

```diff
@@ -81,17 +91,27 @@ public ConnectionManager(IProgram program, Rothko.IOperatingSystem os)
 
         public IConnection CreateConnection(HostAddress address, string username)
         {
-            return new Connection(this, address, username);
+            return SetupConnection(address, username);
         }
 
         public bool AddConnection(HostAddress address, string username)
         {
             if (Connections.FirstOrDefault(x => x.HostAddress == address) != null)
                 return false;
-            Connections.Add(new Connection(this, address, username));
+            Connections.Add(SetupConnection(address, username));
             return true;
         }
 
+        void AddConnection(Uri hostUrl, string username /*, [AllowNull] IEnumerable<RepositoryCacheItem> repositories */)
+        {
+            var address = HostAddress.Create(hostUrl);
+            if (Connections.FirstOrDefault(x => x.HostAddress == address) != null)
+                return;
+            var conn = SetupConnection(address, username);
+            //repositories?.ForEach(r => conn.Repositories.Add(new SimpleRepositoryModel(r.Name, r.CloneUrl, r.LocalPath)));
```

Comment:
Yeah, I need to take this out, I don't think we want to cache this information in our cache file (that's what it was doing)